app/controllers/posts_controller.rb

```diff
@@ -30,6 +30,20 @@ def index
     @posts = Post.where(author_id: @user.id).includes(:comments, :user).paginate(page: params[:page], per_page: 10)
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+
+    authorize! :destroy, @post
+    @author = @post.user
+    @author.decrement!(:posts_counter)
+    @post.destroy
+    if @post.destroy
+      redirect_to user_posts_path(current_user), notice: 'post was successfully deleted.'
+    else
+      redirect_to redirect_url, alert: 'Failed to delete the post.'
+    end
+  end
```

Comment:
Nice job so far😉. An error is being rendered when I try to delete a post😉.
<img width="1371" alt="image" src="https://github.com/maths29/blog_app/assets/64643995/7361c969-700b-40c2-b937-52b39eeac554">

Apart from this, Kindly add the link for deleting a comment😉. This is required to test the delete comment action for admins and users😉. I also noticed that the sign_out button has been removed and I had to manually sign out, for a better user experience and to make it easier for others to test your project, I suggest you add the sign_out button👍🏼.


Lastly, for post-authorization, kindly do well to add ` load_and_authorize_resource` to your post controller😉.